src/lib.rs

```diff
@@ -261,38 +292,38 @@ if_std! {
         fd.seek(SeekFrom::Start(0))?;
         peek_bytes(&bytes)
     }
-}
 
-#[cfg(all(feature = "endian_fd", feature = "elf64", feature = "elf32", feature = "pe64", feature = "pe32", feature = "mach64", feature = "mach32", feature = "archive"))]
```

Comment:
Why is it safe to remove the intersection of all these features here and in the impl ?

Couldn’t the user compile with elf, mach, and std and the get a missing symbol error in the impl parser since PE wasn’t compiled. 

Or does endian_fd mean everything is enabled now ?

Reply:
In https://github.com/m4b/goblin/issues/75#issuecomment-361093100, you said that the parser flag (aka endian_fd) should turn on every other flag. So that is now enforced in Cargo.toml. Since this is all wrapped in `if_endian_fd`, specifying the features here is redundant. As I said in that issue, I'm not sure if this is correct, but I actually kind of like not needing to list out all the features here, so I went with that, but I'm happy to change it.